sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4140,6 +4141,21 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
       }
     }
   }
+
+  test("SPARK-34919: Change partitioning to SinglePartition if partition number is 1") {
```

Comment:
since we don't check the result, it's more like a UT. shall we move it to `PlannerSuite`?